### ISSUE TITLE
Inconsistent Predictions for the Same Ligand-Receptor Pair

### ISSUE THREAD

**Issue by quangnguyenbn99** [BioinfoMachineLearning/FlowDock#6]
Thank you for your interesting work.

I noticed that the predictions for the same ligand-receptor pair vary across multiple runs. Could you clarify why this is happening and how to ensure consistent results?

**Comment by amorehead** [comment]
Hi, @quangnguyenbn99. Thanks for reaching out!

Per your request, I've just added a [`seed`](https://github.com/BioinfoMachineLearning/FlowDock/commit/a180b29962cd27b56b0c4f98cad6fe871f15a7ba) argument for the `sample.py` script that you can customize as `python3 sample.py seed=SEED_VALUE`. This should be a first step towards ensuring more consistent results. However, keep in mind that any PyTorch-based neural network that uses any `scatter` operations (which use atomic operations under the hood) will inevitably contain some degree of stochasticity in their execution. You can try to mitigate this by also setting the Lightning `Trainer`'s `deterministic` flag to `True` following https://lightning.ai/docs/pytorch/stable/common/trainer.html#reproducibility, but this may not always translate to hardware-level determinism.

Let me know if you have any follow-up questions.

**Comment by quangnguyenbn99** [comment]
Hi @amorehead ,
Thanks for the update! Fixing the seed does work, and I'm now getting more consistent results. I appreciate the suggestion!

I also wanted to ask do you have any recommendations on how to accelerate the prediction speed. It seems like in the sample period, the model does not use the GPU resource optimally? Let me know if you have any insights.

Looking forward to your response!

**Comment by amorehead** [comment]
Hi, @quangnguyenbn99. Glad to hear fixing the seed solved your issue. Regarding inference speed, I'd welcome a PR for caching the model weights in GPU memory when performing batch inference for many different inputs. Right now, inference is bottlenecked by needing to reload the model everytime inference is performed.

**Comment by amorehead** [comment]
Hi, @quangnguyenbn99. In the meantime, I will be closing this issue. If you are interested in improving the batched inference speed, please don't hesitate to send me any questions you might have (e.g., in a new PR)!
